src/postformatting/unused-fontface-remover.js

```diff
@@ -1,59 +1,77 @@
-          fontNameValues.push(value.value)
-        }
-      })
-    } else if (rule.type === 'Atrule' && rule.name === 'media') {
-      rule.block.children.forEach(handleRule)
-    }
+function decodeFontName (node) {
+  let name = csstree.generate(node)
+  // TODO: use string decode
+  if (name[0] === '"' || name[0] === "'") {
+    name = name.substr(1, name.length - 2)
   }
-  rules.forEach(handleRule)
+  return name
+}
+
+function getAllFontNameValues (ast) {
+  const fontNameValues = new Set()
+
+  csstree.walk(ast, {
+    visit: 'Declaration',
+    enter: function (node) {
+      // walker pass through `font-family` declarations inside @font-face too
+      // this condition filter them, to walk through declarations inside a rules only
+      if (this.rule) {
+        csstree.lexer
+          .findDeclarationValueFragments(node, 'Type', 'family-name')
+          .forEach(entry => {
+            const familyName = decodeFontName({
+              type: 'Value',
+              children: entry.nodes
+            })
+            debuglog('found used keyframe animation: ' + familyName)
+            fontNameValues.add(familyName)
+          })
+      }
+    }
+  })
+
   return fontNameValues
 }
 
-function unusedFontfaceRemover (rules) {
+function unusedFontfaceRemover (ast) {
   debuglog('getAllFontNameValues')
   // NOTE: we grabp the full declaration everywhere a font name is used,
   // and just do a simple index of on these lines to see if each @font-face
   // is used anywhere. Could in theory yield false positives, but is quite unlikely,
-  // unless people use css keywoards in their custom font names.
-  const fontNameValues = getAllFontNameValues(rules)
+  // unless people use css keywords in their custom font names.
+  const fontNameValues = getAllFontNameValues(ast)
   debuglog('getAllFontNameValues AFTER')
 
-  function filterUnusedFontFaceRule (rule) {
-    if (!(rule.type === 'Atrule' && rule.name === 'font-face')) {
-      return true
-    }
-    // was this @font-face used?
-    return rule.block.children.some(({ property, value }) => {
-      let toKeep = false
-      let font
-      if (property === 'font-family') {
-        font = value.value
-          // replace quoute mark to ensure we match inside font declaration,
-          // in case appears without them there
-          .replace(/^['"]/, '')
-          .replace(/['"]$/, '')
+  // remove all unused font-face declarations
+  csstree.walk(ast, {
+    visit: 'Atrule',
+    enter: (atrule, atruleItem, atruleList) => {
+      if (csstree.keyword(atrule.name).basename !== 'font-face') {
+        return
       }
-      if (font) {
-        toKeep = fontNameValues.some(value => value.indexOf(font) !== -1)
-        if (!toKeep) {
-          debuglog('drop unused font-face rule: ' + font)
+
+      csstree.walk(atrule, {
+        visit: 'Declaration',
+        enter: declaration => {
+          if (csstree.property(declaration.property).name === 'font-family') {
```

Comment:
Add your new font-face check here instead so you don't have to `walk` one extra time for it (if no `src`, remove).